packages/app/src/debug/DebugPageHeader.cy.tsx

```diff
@@ -32,27 +34,70 @@ describe('<DebugPageHeader />', {
     })
 
     cy.findByTestId('debug-header').children().should('have.length', 2)
-    cy.findByTestId('debug-test-summary')
-    .should('have.text', 'Adding a hover state to the button component')
+    cy.findByTestId('debug-test-summary').should('have.text', 'Adding a hover state to the button component')
 
-    cy.findByTestId('debug-runCommit-info').children().should('have.length', 3)
-    cy.findByTestId('debug-runNumber')
-    .should('have.text', ' Run #432')
-    .should('have.css', 'color', 'rgb(90, 95, 122)')
-
-    cy.findByTestId('debug-commitsAhead')
-    .should('have.text', 'You are 2 commits ahead')
-    .should('have.css', 'color', 'rgb(189, 88, 0)')
+    cy.findByTestId('debug-commitsAhead').should('have.text', 'You are 2 commits ahead')
 
     cy.findByTestId('debug-results').should('be.visible')
 
+    cy.findByTestId('debug-runNumber-FAILED')
+    .should('have.text', '#432')
+    .children().should('have.length', 2)
+
+    cy.findByTestId('debug-flaky-badge')
+    .should('not.exist')
+
     defaults.forEach((obj) => {
       cy.findByTestId(obj.attr)
       .should('have.text', obj.text)
       .children().should('have.length', 2)
     })
   })
 
+  it('displays a flaky badge', () => {
+    const flakyCount = 4
+
+    cy.mountFragment(DebugPageFragmentDoc, {
+      onResult: (result) => {
+        if (result) {
+          result.totalFlakyTests = flakyCount
+        }
+      },
+      render: (gqlVal) => {
+        return (
+          <DebugPageHeader gql={gqlVal} commitsAhead={0}/>
+        )
+      },
+    })
+
+    cy.findByTestId('debug-flaky-badge')
+    .contains(defaultMessages.specPage.flaky.badgeLabel)
+
+    cy.findByTestId('total-flaky-tests')
+    .contains(flakyCount)
+  })
+
+  it('checks if all run status are accounted for', () => {
+    const statuses: CloudRunStatus[] = ['PASSED', 'FAILED', 'CANCELLED', 'RUNNING', 'ERRORED']
+
+    statuses.forEach((status) => {
+      cy.mountFragment(DebugPageFragmentDoc, {
+        onResult: (result) => {
+          if (result) {
+            result.status = status
+          }
+        },
+        render: (gqlVal) => {
+          return (
+            <DebugPageHeader gql={gqlVal} commitsAhead={0}/>
+          )
+        },
+      })
+
+      cy.findByTestId(`debug-runNumber-${status}`).should('be.visible')
```

Comment:
My reason for adding the percy snapshots was because each `status` has a unique color and icon hence in order to visually compare all the possible combinations of `status` pairings without explicitly testing the colors through `rgb` values I added the percy snapshots.
I could see this either way (with or without the snapshots) so I'm happy to remove it if that is the consensus

Reply:
@amehta265 I'm confused - I don't see Percy snapshots being taken in this test currently. I'm suggesting that we add them. Am I missing something?

Reply:
If there is a unique UI state, then we should try and capture it with a percySnapshot! (In response to your first question)

Reply:
I've added the percySnaphot!